Measurements/Measurements_Workbook.ipynb

```diff
@@ -0,0 +1,453 @@
+    "The state of the qubits at the end of the operation does not matter."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Solution**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    " operation BasisStateMeasurement_Reference (qs : Qubit[]) : Int {\n",
+    "        // Measurement on the first qubit gives the higher bit of the answer, on the second - the lower\n",
+    "        let m1 = M(qs[0]) == Zero ? 0 | 1;\n",
+    "        let m2 = M(qs[1]) == Zero ? 0 | 1;\n",
+    "        return m1 * 2 + m2;\n",
+    "    }"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In task 1.5 wrote an operation to measure for the states $|00\\rangle$ and $|11\\rangle$, in this task we need to account for the mix-states $|01\\rangle$ and $|10\\rangle$. This time measuring the first qubit won't give use any information on the second qubit, so we need to measure both qubits. \n",
+    "m1 store the state of the first qubit and m2 store the state of the second qubit. \n",
+    "\n",
+    "Follow 4 possible outputs: \n",
+    "- m1 is $|0\\rangle$ and m2 is $|0\\rangle$, we return 0*2+0 = 0\n",
+    "- m1 is $|0\\rangle$ and m2 is $|1\\rangle$, we return 0*2+1 = 1\n",
+    "- m1 is $|1\\rangle$ and m2 is $|0\\rangle$, we return 1*2+0 = 2\n",
+    "- m1 is $|1\\rangle$ and m2 is $|1\\rangle$, we return 1*2+1 = 3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.7. Distinguish two basis states given by bit strings\n",
+    "\n",
+    "**Inputs:**\n",
+    "1.\tN qubits (stored in an array of length 𝑁) which are guaranteed to be in one of the two basis states described by the given bit strings.\n",
+    "2.\tTwo bit strings represented as Bool[]s.\n",
+    "**Output:**\n",
+    "•\t0 if the qubits were in the basis state described by the first bit string,\n",
+    "•\t1 if they were in the basis state described by the second bit string.\n",
+    "Bit values false and true correspond to $|0\\rangle$ and $|1\\rangle$ states. You are guaranteed that both bit strings have the same length as the qubit array, and that the bit strings differ in at least one bit.\n",
+    "You can use exactly one measurement. The state of the qubits at the end of the operation does not matter.\n",
+    "Example: for bit strings [false, true, false] and [false, false, true] return 0 corresponds to state $|010\\rangle$ and return 1 corresponds to state $|001\\rangle$.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Solution**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "function FindFirstDiff_Reference (bits1 : Bool[], bits2 : Bool[]) : Int {\n",
```

Comment:
Q# notebooks don't support combining `%kata` magic with multiple functions/operations defined in a single cell, so you'll need to split this cell into two - one with the definition of `FindFirstDiff_Reference` and another one with `%kata` and the definition of the operation `TwoBitstringsMeasurement` itself.